assets/onboarding/index.jsx

```diff
@@ -1,24 +1,28 @@
-import { Stepper } from '@woocommerce/components';
 import { render } from '@wordpress/element';
-import { useFullScreen } from './use-fullscreen';
+import { useFullScreen } from '../react-hooks';
 import { steps } from './steps.jsx';
+import { QueryStringRouter } from './query-string-router';
+import Navigation from './navigation';
+import ContentContainer from './content-container';
 
-function SenseiOnboardingPage() {
-	useFullScreen( [ 'sensei-color', 'sensei-onboarding__page' ] );
+/**
+ * Query string name used to route the onboarding wizard.
+ */
+const QUERY_STRING_NAME = 'onboarding-step';
 
-	const currentStep = steps[ 0 ];
+const SenseiOnboardingPage = () => {
+	useFullScreen( [ 'sensei-color', 'sensei-onboarding__page' ] );
 
 	return (
-		<>
-			<div className="sensei-onboarding__header">
-				<Stepper currentStep={ currentStep.key } steps={ steps } />
-			</div>
-			<div className="sensei-onboarding__container">
-				{ currentStep.container }
-			</div>
-		</>
+		<QueryStringRouter
+			routes={ steps }
+			queryStringName={ QUERY_STRING_NAME }
+		>
+			<Navigation routes={ steps } />
+			<ContentContainer />
+		</QueryStringRouter>
```

Comment:
I think the layout classes, `sensei-onboarding__header`, `sensei-onboarding__container` should remain here and wrap the `Navigation` and `ContentContainer` components, as these describe the structure of the page, and don't really belong inside those components.

Reply:
Sounds good!
https://github.com/Automattic/sensei/pull/3085/commits/cfabe943b5d90dc7fb7eddfd993732d6ee2a707e